src/mono_font/draw_target.rs

```diff
@@ -0,0 +1,140 @@
+use crate::{
+    draw_target::DrawTarget, geometry::Dimensions, iterator::ContiguousIteratorExt,
+    pixelcolor::BinaryColor, primitives::Rectangle, Pixel,
+};
+
+pub struct MonoFontDrawTarget<'a, T, C> {
+    parent: &'a mut T,
+    colors: C,
+}
+
+impl<'a, T: DrawTarget, C> MonoFontDrawTarget<'a, T, C> {
```

Comment:
I think you lost me here. Is this a DrawTarget that limits or extends the allowed operations? Both sound weird to me. The difference between the three implementations is the colors used, which I think would have a better place in a paint-like object.

Reply:
> I think you lost me here. Is this a DrawTarget that limits or extends the allowed operations?

Maybe the name `MonoFontDrawTarget` is a bit confusing. `MonoFontDrawTarget` is a draw target adapter similar to the `ColorConverted` draw target adapter:
https://github.com/embedded-graphics/embedded-graphics/blob/aa5bd3a2ef67d85669cd92b1a18fd1cd253edc53/src/draw_target/mod.rs#L155-L215
It is only used internally and doesn't affect the external API. In contrast to the generic `ColorConverted` adapter, `MonoFontDrawTarget` can convert the draw calls made by a `BinaryColor` image drawable into different draw calls, depending on the colors which are set in a `MonoTextStyle`. If only one color (text or bg) is set it will call `draw_iter` on the parent target and if both colors are set it can call the more efficient `fill_contiguous`. To reduce the runtime overhead for this distinction there are three different impls of `MonoFontDrawTarget`.

> The difference between the three implementations is the colors used, which I think would have a better place in a paint-like object.

The advantage of implementing the color conversion as a draw target adapter is that the font rendering becomes independent of the image drawable implementation. While `MonoFont` currently only supports `ImageRaw`s to store the glyph image it wouldn't be hard to extend this to other image formats in the future. This would make it possible to load fonts from a BMP file or a custom compressed storage  format. But it would add a type parameter to `MonoFont`, which this PR tried to get rid of, so that I opted not to implement this at the moment.

> At this point I'm actually more worried about embedded-gui. This way I don't think it'll be possible to use two different font sets in a gui? Or maybe I'm missing something atm, like labels creating their own draw targets.

What do you mean by "font sets"? Fonts from different font renderers?

Reply:
> What do you mean by "font sets"? Fonts from different font renderers?

Yes. But I think I simply misunderstood what this PR means. That's the curse of a cursory glance. You're saying this is internal only so I guess my worries are invalid. I'll need to put in more time to check things, though.

Thanks for the info!